File: InsuraTech/InsuraTech.Services/Database/InsurancePolicy.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace InsuraTech.Services.Database
{
    public partial class InsurancePolicy:BaseEntity
    {
        public int InsurancePolicyId { get; set; }

        public int InsurancePackageId { get; set; }
        [ForeignKey("InsurancePackageId")]
        public virtual InsurancePackage InsurancePackage { get; set; } = null!;

        public int ClientId { get; set; }
        [ForeignKey("ClientId")]
        public virtual Client Client { get; set; } = null!;

        public DateTime StartDate { get; set; }

        public DateTime EndDate { get; set; }

        public bool IsActive { get; set; }
        public bool HasActiveClaimRequest { get; set; }
        public bool IsNotificationSent { get; set; }
        public bool IsPaid { get; set; }
        public ICollection<ClaimRequest> ClaimRequests { get; set; } = new List<ClaimRequest>();
        public ICollection<Transaction> Transactions { get; set; } = new List<Transaction>();
    }

}

File: InsuraTech/UI/insuratech_mobile/lib/models/insurance_policy.dart
import 'package:insuratech_mobile/models/client.dart';
import 'package:insuratech_mobile/models/client_feedback.dart';
import 'package:insuratech_mobile/models/insurance_package.dart';
import 'package:json_annotation/json_annotation.dart';
part 'insurance_policy.g.dart';

@JsonSerializable()
class InsurancePolicy {
  int? insurancePolicyId;
  int? insurancePackageId;
  int? clientId;
  String? startDate;
  String? endDate;
  bool? isActive;
  bool? isNotificationSent;
  bool? isPaid;
  InsurancePackage? insurancePackage;
  Client? client;
  bool? hasActiveClaimRequest;
  ClientFeedback? clientFeedback;

  InsurancePolicy();

  factory InsurancePolicy.fromJson(Map<String, dynamic> json) =>
      _$InsurancePolicyFromJson(json);

  /// Connect the generated [_$PersonToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$InsurancePolicyToJson(this);
}
